ls1intum/Artemis
# `Adaptive learning`: Load legend of learning path graph dynamically

### Checklist
#### General



#### Client

### Motivation and Context



The legend of the learning path graph should display only descriptions for nodes that are actually displayed.

Resolves #7494 .

### Description


With this PR we check for the presence of node types and display the description in the legend only once the corresponding type is currently displayed.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- Course with Competencies (Linked to lecture units and exercises) and Learning Paths enabled

1. Log in to Artemis
2. View the learning path graph, the legend should contain only the entries for exercises and competency end nodes
3. Navigate to Competency Management and add a matching relation between two competencies
4. View the learning path graph, the legend should now also contain the entries for matching start (branching) and end (dot)


### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage







#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [learning-path-graph.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5694/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/learning-paths/learning-path-graph/learning-path-graph.component.ts.html) | 87.93% | ✅  |
| [learning-path-legend.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5694/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/learning-paths/learning-path-graph/learning-path-legend.component.ts.html) | 100% | ✅ |

�

### Screenshots


![Screenshot 2023-11-07 at 14 23 57](https://github.com/ls1intum/Artemis/assets/44003963/d15a723b-9750-44dc-a9bc-a62e6e6d6062)

![Screenshot 2023-11-07 at 14 24 57](https://github.com/ls1intum/Artemis/assets/44003963/bac5021d-dd70-4355-b17f-3adc063bdbc9)
